### PR TITLE
(PE-31686) Ship puppet 6.23.0 with bolt/ace server

### PR DESCRIPTION
**Pull request by donoghuc**
Take up API changes for ace server to use v4 catalog endpoint more efficently.

### CODE REVIEW

**Review by steveax — APPROVED** [review]
👍

File: configs/components/rubygem-puppet.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
 component 'rubygem-puppet' do |pkg, settings, platform|
   # Projects may define a :rubygem_puppet_version setting, or we use 6.20.0 by default:
-  version = settings[:rubygem_puppet_version] || '6.20.0'
```
Comment (ekinanp) [inline configs/components/rubygem-puppet.rb:3]: Hmm, any reason why we're not calculating this from `rubygem_puppet_version` ?

Comment (donoghuc) [reply configs/components/rubygem-puppet.rb:3]: I figured we would do that work when we pull in the puppet7 in our components ticket. same as the gemspec change in the ace repo.

Comment (donoghuc) [reply configs/components/rubygem-puppet.rb:3]: oh, hmm seems we need a ticket for that https://tickets.puppetlabs.com/browse/PE-31684

Comment (donoghuc) [reply configs/components/rubygem-puppet.rb:3]: oh, nevermind, found it https://tickets.puppetlabs.com/browse/PE-31696

Comment (donoghuc) [reply configs/components/rubygem-puppet.rb:3]: as part of that ticket we will move to setting :rubygem_puppet_version in the individual bolt-runtime project definitions.

**Pull request merged**